src/PSR7/Validators/ArrayValidator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace League\OpenAPIValidation\PSR7\Validators;
+
+use cebe\openapi\spec\Parameter;
+use League\OpenAPIValidation\PSR7\Exception\Validation\InvalidParameter;
+use League\OpenAPIValidation\PSR7\Exception\Validation\RequiredParameterMissing;
+use League\OpenAPIValidation\PSR7\OperationAddress;
+use League\OpenAPIValidation\Schema\BreadCrumb;
+use League\OpenAPIValidation\Schema\Exception\SchemaMismatch;
+use League\OpenAPIValidation\Schema\SchemaValidator;
+use function array_key_exists;
+
+/**
+ * Validates given array against given specs
+ */
+class ArrayValidator
+{
+    /** @var Parameter[] */
+    private $specs;
+
+    /**
+     * @param Parameter[] $specs
+     */
+    public function __construct(array $specs)
+    {
+        $this->specs = $specs;
+    }
+
+    /**
+     * @param mixed[] $params
+     *
+     * @throws InvalidParameter
+     * @throws RequiredParameterMissing
+     */
+    public function validateArray(OperationAddress $addr, array $params, int $validationStrategy) : void
+    {
+        foreach ($this->specs as $name => $spec) {
+            if ($spec->required && ! array_key_exists($name, $params)) {
+                throw RequiredParameterMissing::fromNameAndAddr($name, $addr);
```

Comment:
Not really important but I generally prefer do a rethrow outside method with `$addr` instead of passing it to the method for only be used to construct an exception. This make further code readers think that some logic depends on `$addr` inside this method. This class look like low-level utility class and with moving `$addr` and PSR-7 specific exception could even live in `League\OpenAPIValidation\Schema` namespace

Reply:
That's definitely a redundancy, I haven't noticed it